File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
      unordered_map<int,int> mp;
        int n=nums.size();
        int res=0;
        for(auto &x:nums)
        {
            res+=mp[x]++;
        }
        return res;
    }
};

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.py
class Solution(object):
    def numIdenticalPairs(self, nums):
        
        n=len(nums)
        count=0
        
        for i in range(n):
            for j in range(i+1,n):
                if nums[i] == nums[j]:
                    count+=1
        return count
                
        